src/js/containers/explorer/detail/table/ExplorerTableContainer.jsx

```diff
@@ -0,0 +1,259 @@
+        // Change page number in Redux state
+        if (inRange) {
+            this.props.setExplorerTablePage(pageNumber);
+        }
+    }
+
+    getPager(totalItems, currentPage, pageSize) {
+        // calculate total pages
+        const totalPages = Math.ceil(totalItems / pageSize);
+
+        let startPage;
+        let endPage;
+        let prevEllipses = (<span className="pagination-ellipsis">...</span>);
+        let nextEllipses = (<span className="pagination-ellipsis">...</span>);
+        let firstButton = (
+            <li>
+                <button onClick={() => this.onChangePage(1, true)}>{1}</button>
+            </li>
+        );
+        let lastButton = (
+            <li>
+                <button onClick={() => this.onChangePage(totalPages, true)}>{totalPages}</button>
+            </li>
+        );
+        if (totalPages < 5) {
+            // less than 5 total pages so show all
+            startPage = 1;
+            endPage = totalPages;
+            prevEllipses = '';
+            nextEllipses = '';
+            firstButton = '';
+            lastButton = '';
+        }
+        else {
+            if (currentPage === 1) {
+                startPage = currentPage;
+                endPage = currentPage + 2;
+            }
+            else if (currentPage === totalPages) {
+                startPage = currentPage - 2;
+                endPage = currentPage;
+            }
+            else {
+                startPage = currentPage - 1;
+                endPage = currentPage + 1;
+            }
+
+            if (currentPage < 4) {
+                prevEllipses = '';
+                firstButton = '';
+            }
+            else if (currentPage > (totalPages - 3)) {
+                nextEllipses = '';
+                lastButton = '';
+            }
+        }
+
+        // calculate start and end item indexes
+        const startIndex = (currentPage - 1) * pageSize;
+        const endIndex = Math.min(startIndex + (pageSize - 1), (totalItems - 1));
+
+        // create an array of pages to repeat in the pager control
+        const pages = range(startPage, endPage + 1);
+
+        // return object with all pager properties
+        return {
+            totalItems,
+            currentPage,
+            pageSize,
+            totalPages,
+            startPage,
+            endPage,
+            startIndex,
+            endIndex,
+            pages,
+            prevEllipses,
+            nextEllipses,
+            firstButton,
+            lastButton
+        };
+    }
+
+    setPageOfItems() {
+        const page = this.props.pageNumber;
+        const results = this.state.results;
+
+        // Get new pager object for specified page
+        const pager = this.getPager(results.length, page, 20);
+
+        // Get new page of items from results
+        const pageOfItems = results.slice(pager.startIndex, pager.endIndex + 1);
+
+        // update state
+        this.setState({
+            pager,
```

Comment:
Avoid storing React components in state. React evaluates inbound state/props for changes from the current set of state/props to determine if it needs to re-render. However, if components are stored in state, a comparison can't be made so a render happens by default. A better approach is to precalculate a complex child component's data and store it as an object that the child component can parse, compare, and display.